### DISCUSSION TITLE
maya-usd changing the hierarchy when exporting

### DISCUSSION THREAD

**Discussion by flord** [Autodesk/maya-usd#3635] [General]
Hi,
One of the new important rules when working with a USD pipeline is to never change the hierarchy of elements. Riggers are hit particularly hard by this one: they used to change all they wanted in the hierarchy of the incoming modeling in orter to suit their needs, but this is now illegal in a USD pipeline because the animation bakes in shots are now combined with the assets (containing modeling and shading) and if the hierarchies don't match, not only the materials will not be properly assigned, but meshes will be duplicated. 

To our surprise, we discovered that maya-usd modifies the hierarchy when exporting the anim bake when a mesh has a constraint in the rig. Here is an example.
This is part of a hierarchy as it comes out from modeling. 
![20240226-154559](https://github.com/Autodesk/maya-usd/assets/575125/bc8a4627-e877-43d5-bf1c-9c9573d15395)

And this is what it look like when we export the animation from the rigged character.
![20240226-154614](https://github.com/Autodesk/maya-usd/assets/575125/376929c2-19ed-4b4d-8714-c3354e26f6bd)

And this is the asset rigged in Maya. We export from `serge_geometry_defVar_hi` down since we don't need the rig itself in the anim bake.
![20240226-154715](https://github.com/Autodesk/maya-usd/assets/575125/837bb525-d5bd-4a9a-b42c-2501275561e3)

I understand that Maya changes the hierarchy internally to accomodate for the constraint, but we really don't need those extra prims when exporting the hierarchy. And prims that were meshes in one layer become Xforms in the other. Only the meshes with constraints exhibit this problem, the meshes that are skinned in the rig do export normally with the same hierarchy as the input. We use the `mergeTransformAndShape=1` option when exporting.

As a workaround for now, we can create groups for objects that are constrainted, but it creates needless back-and-forth communications between the modeler and the rigger. 

In our opinion, this behavior should be fixed.
Thanks!

F

**Comment by wallworm** [comment]
Thank you for sharing. We will discuss this internally and get back to you.
